src/Plutarch/Extra/IsData.hs

```diff
@@ -105,11 +118,19 @@ gProductFromBuiltinData (BuiltinData (List xs)) = do
     productTypeTo <$> hctraverse (Proxy @FromData) (unI . mapKI fromData) prod
 gProductFromBuiltinData _ = Nothing
 
-{- |
-  Unsafe version of 'gProductFromBuiltinData'.
+-- | @since 1.4.0
+newtype DerivePDataFieldsViaDataList (a :: S -> Type) (s :: S)
+    = DerivePDataFieldsViaDataList (a s)
 
-  @since 1.1.0
--}
+-- | @since 1.4.0
+instance
+    (PDataFields (PInner a), PInner (DerivePDataFieldsViaDataList a) ~ a) =>
+    PDataFields (DerivePDataFieldsViaDataList a)
+    where
+    type PFields (DerivePDataFieldsViaDataList a) = PFields (PInner a)
+    ptoFields x = ptoFields $ pto $ pto x
```

Comment:
This should work just fine, but I wonder if you'd be able to use IsWrappedType from SOP.Generic :thinking: